endpoints/status.go

```diff
@@ -6,9 +6,17 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
-// Status is the handler function of the "/status" endpoint
-func Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	// We might want more logic here eventually... but for now, we're ok to serve more traffic as
-	// long as the server responds.
-	w.WriteHeader(http.StatusNoContent)
+// NewStatusEndpoint returns a handler which writes the given response when the app is ready to serve requests.
+func NewStatusEndpoint(response string) httprouter.Handle {
+	// Today, the app always considers itself ready to serve requests.
+	if response == "" {
+		return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+			w.WriteHeader(http.StatusNoContent)
+		}
+	}
+
+	responseBytes := []byte(response)
+	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+		w.Write(responseBytes)
```

Comment:
Does this result in a 200 status code automatically?

Reply:
Yes, it does. From Go doc for this function:
```
func (http.ResponseWriter).Write([]byte) (int, error)
Write writes the data to the connection as part of an HTTP reply.

If WriteHeader has not yet been called, Write calls WriteHeader(http.StatusOK) before writing the data
```

Reply:
Yes, it does. From Go doc for this function:
```
func (http.ResponseWriter).Write([]byte) (int, error)
Write writes the data to the connection as part of an HTTP reply.

If WriteHeader has not yet been called, Write calls WriteHeader(http.StatusOK) before writing the data
```